src/action/hostname.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the hostname action.
+
+use crate::session::{self, Session};
+#[cfg(target_family = "unix")]
+use libc::c_char;
+#[cfg(target_family = "unix")]
+use std::ffi::CStr;
+use std::ffi::OsString;
+#[cfg(target_family = "unix")]
+use std::io::Error;
+#[cfg(target_family = "unix")]
+use std::os::unix::prelude::OsStringExt as _;
+#[cfg(target_os = "windows")]
+use std::os::windows::ffi::OsStringExt as _;
+#[cfg(target_os = "windows")]
+use windows::{
+    core::{Error, PWSTR},
+    Win32::System::SystemInformation::{ComputerNamePhysicalDnsHostname, GetComputerNameExW},
+};
```

Comment:
You can scope imports under the functions that actually use them. Otherwise there is a bit of a mess, e.g. `Error` refers to completely different types depending on the platform (`std::io::Error` on Linux, `std::windows::core::Error` on Windows). Sometimes it is useful but I think it just increases confusion here.

Reply:
Yes, I was not sure what the general coding style is. In general, I prefer to scope imports too.